pkg/kube/pod.go

```diff
@@ -82,10 +82,9 @@ func (pd *podClient) DeletePod(namespace, name string) error {
 
 func (pd *podClient) ListPods(namespace string, selector labels.Selector) ([]corev1.Pod, error) {
 	podList := &corev1.PodList{}
-	if err := pd.kubecli.List(context.TODO(), podList, &client.ListOptions{LabelSelector: selector, Namespace: namespace}); err != nil {
-		if err != nil {
-			return nil, err
-		}
+	err := pd.kubecli.List(context.TODO(), podList, &client.ListOptions{LabelSelector: selector, Namespace: namespace})
+	if err != nil {
```

Comment:
The `if` doesn't seem to be redundant.

And we need to reduce the scope of variables,  references from [Uber go style: Reduce Scope of Variables](https://github.com/uber-go/guide/blob/master/style.md#reduce-scope-of-variables).

Reply:
What about this:

```go
if err := pd.kubecli.List(context.TODO(), podList, &client.ListOptions{LabelSelector: selector, Namespace: namespace}); err != nil {
    return nil, err
}
```

What I wanted to remove is the nested `if`.

Reply:
Besides, I wonder the necessity of reducing the scope of `err` in this case. What is the difference between these two styles?

Reply:
Emm, u did delete the nested `if`. And what you mentioned above is very good.

The style varies from person to person,  and this repo follows the Uber style.
In addition, I think it‘s more concise and will not cause other problems due to `err` coverage.

Reply:
Ok, consistency is what matters, and I'll follow this style.

Reply:
I don't think a local variable is more concise than a "global" one.

Since we always handle errors and we handle every `err` case first, I think it's rare to encounter problems caused by the `err` coverage.

Reply:
First, thank you for your understanding.
And I can't find a reason to explain why we have to do it, but we found that most of gopher did it.

Reply:
I updated this PR to follow the Uber style.

PTAL.